src/Juvix/Prelude/Base.hs

```diff
@@ -223,8 +223,15 @@ tableNestedInsert k1 k2 = tableInsert (HashMap.singleton k2) (HashMap.insert k2)
 --------------------------------------------------------------------------------
 
 revAppend :: [a] -> [a] -> [a]
-revAppend [] ys = ys
-revAppend (x : xs) ys = revAppend xs (x : ys)
+revAppend [] !ys = ys
+revAppend (x : xs) !ys = revAppend xs (x : ys)
+
+map' :: (a -> b) -> [a] -> [b]
+map' _ [] = []
+map' f (h : t) =
+  let !v = f h
+   in let !vs = map' f t
```

Comment:
minor: combine lets:

```
let !v = f h
    !vs = map' f t
in v : vs
```

Reply:
No! This changes the semantics. GHC is free to first evaluate `vs` and then `v` if you combine them (and this is actually what happens with my GHC).

Reply:
oh I didn't know that - thanks 👍

Reply:
Good idea ! Someone could subtly refactor this